pilot/pkg/model/sidecar.go

```diff
@@ -663,6 +664,36 @@ func serviceMatchingListenerPort(service *Service, ilw *IstioEgressListenerWrapp
 	return nil
 }
 
+func serviceMatchingVirtualServicePorts(service *Service, vsDestPorts sets.IntSet) *Service {
+	// A value of 0 in vsDestPorts is used as a sentinel to indicate a dependency
+	// on every port of the service.
+	if len(service.Ports) == 1 || len(vsDestPorts) == 0 || vsDestPorts.Contains(0) {
+		return service.DeepCopy()
```

Comment:
```suggestion
		return service
```

Can we either not copy or comment why we need to copy? Its not clear to me its needed/why